packages/integration-sdk-cli/src/neo4j/neo4jGraphStore.ts

```diff
@@ -0,0 +1,168 @@
+import { Entity, Relationship } from '@jupiterone/integration-sdk-core';
+
+import * as neo4j from 'neo4j-driver';
+
+export interface Neo4jGraphObjectStoreParams {
+  uri: string;
+  username: string;
+  password: string;
+}
+
+export class Neo4jGraphStore {
+  private neo4jDriver: neo4j.Driver;
+  private persistedSession: neo4j.Session;
+  private databaseName = 'neo4j';
+  private typeList = new Set<string>();
+  private integrationInstanceID: string;
+
+  constructor(instanceID: string, params: Neo4jGraphObjectStoreParams, session?: neo4j.Session) {
+    if(session) {
+      this.persistedSession = session;
+    }
+    else {
+      this.neo4jDriver = neo4j.driver(
+        params.uri,
+        neo4j.auth.basic(params.username, params.password),
+      );
+    }
+    this.integrationInstanceID = instanceID;
+  }
+
+  private async runCypherCommand(cypherCommand: string): Promise<neo4j.Result> {
+    if(this.persistedSession) {
+      const result = await this.persistedSession.run(cypherCommand);
+      return result;
+    }
+    else {
+      const session = this.neo4jDriver.session({
+        database: this.databaseName,
+        defaultAccessMode: neo4j.session.WRITE,
+      });
+      const result = await session.run(cypherCommand);
+      await session.close();
+      return result;
+    }
+  }
+
+  private startsWithNumeric(str: string){
+    return /^\d+$/.test(str);
+  }
+
+  private sanitizePropertyName(propertyName: string): string {
+    let sanitizedName = '';
+    if(this.startsWithNumeric(propertyName)) {
+      sanitizedName += 'n';
+    }
+    sanitizedName += propertyName;
+    sanitizedName = sanitizedName.replace(/[\s!@#$%^&*()-=+\\|'";:/?.,><`~\t\n[\]{}]/g, "_");
+    return sanitizedName;
+  }
+
+  private sanitizeValue(value: string): string {
+    return value.replace(/'/gi, "\\'")
+  }
+
+  public buildPropertyString(propList, nodeName) {
+    let propString = '';
+    for (const key in propList) {
+      if (key === '_rawData') {
+        //stringify JSON in rawData so we can store it.
+        propString += `${nodeName}.${key} = '${JSON.stringify(propList[key])}', `;
+      } else {
+        // Sanitize out characters that aren't allowed in property names
+        const propertyName = this.sanitizePropertyName(key);
+
+        //If we're dealing with a number or boolean, leave alone, otherwise
+        //wrap in single quotes to convert to a string and escape all
+        //other single quotes so they don't terminate strings prematurely.
+        if(typeof propList[key] == 'number' || typeof propList[key] == 'boolean') {
```

Comment:
@ndowmon after we talked, it got me thinking more about how I'm handling property values and I decided to make a small change to keep number and boolean values as-is and not potentially modify them to be strings.  Low importance, since this will really only impact people as they start performing more complex queries within the neo4j database, but I'm happier with this than what I had before.  That said, if this seems unnecessary/excessive/bad/etc. let me know and I can revert.